### PR TITLE
docs: Improve error handling docs for server-side errors.

### PR DESCRIPTION
**Pull request by delbaoliveira**
This PR adds more info about `error.message`, `error.digest`, and omission of sensitive error details from Server Components.

### CODE REVIEW

**Review by dferber90 — APPROVED** [review]
Thanks!

**Review by williameliasson-dev — APPROVED** [review]

File: docs/02-app/01-building-your-application/01-routing/07-error-handling.mdx [hunk 1]
```diff
@@ -210,6 +213,14 @@ export default function GlobalError({ error, reset }) {
...
-When running `next dev`, the `error` will be serialized and forwarded from the Server Component to the client `error.js`. To ensure security when running `next start` in production, a generic error message is forwarded to `error` along with a `.digest` which contains a hash of the error message. This hash can be used to correspond to server logs.
+#### Securing Sensitive Error Information
+
+During production, the `Error` instance forwarded to the client only includes a generic `message` and `digest` property.
+
+This is a security precaution to avoid leaking potentially sensitive details included in the error to the client.
+
+The `message` property contains a generic message about the error and the `digest` property contains an automatically generated hash of the error that can be used to match the corresponding error in server-side logs.
+
+During development, the `Error` instance will be serialized and forwarded from the Server Component to the `error.js` Client Component.
```
Comment (timneutkens) [inline docs/02-app/01-building-your-application/01-routing/07-error-handling.mdx:226]: To clarify this it'll still be a new error object, but it won't hide the `message` in order to allow you to debug.

Comment (delbaoliveira) [reply docs/02-app/01-building-your-application/01-routing/07-error-handling.mdx]: I've updated to be more specific.

**Review by leerob — DISMISSED** [review]

File: docs/02-app/01-building-your-application/01-routing/07-error-handling.mdx [hunk 2]
```diff
@@ -210,6 +213,14 @@ export default function GlobalError({ error, reset }) {
...
-When running `next dev`, the `error` will be serialized and forwarded from the Server Component to the client `error.js`. To ensure security when running `next start` in production, a generic error message is forwarded to `error` along with a `.digest` which contains a hash of the error message. This hash can be used to correspond to server logs.
+#### Securing Sensitive Error Information
+
+During production, the `Error` object forwarded to the client only includes a generic `message` and `digest` property.
+
+This is a security precaution to avoid leaking potentially sensitive details included in the error to the client.
+
+The `message` property contains a generic message about the error and the `digest` property contains an automatically generated hash of the error that can be used to match the corresponding error in server-side logs.
+
+During development, the `Error` object forwarded to the client will be serialized and include the properties of the original error for easier debugging.
```
Comment (timneutkens) [inline docs/02-app/01-building-your-application/01-routing/07-error-handling.mdx:226]:
```suggestion
During development, the `Error` object forwarded to the client will be serialized and include the `message` of the original error for easier debugging.
```

It doesn't serialize all properties, only `message`.

**Review by leerob — APPROVED** [review]

**Pull request merged**